internal/config/types.go

```diff
@@ -54,6 +60,48 @@ type GDGAppConfiguration struct {
 	Global        *AppGlobals                  `mapstructure:"global" yaml:"global"`
 }
 
+type UserSettings struct {
+	RandomPassword bool `mapstructure:"random_password" yaml:"random_password"`
+	MinLength      int  `mapstructure:"min_length" yaml:"min_length"`
+	MaxLength      int  `mapstructure:"max_length" yaml:"max_length"`
+}
+
+func (u *UserSettings) GetPassword(username string) string {
+	if !u.RandomPassword {
+		return u.defaultUserPassword(username)
+	} else if u.MinLength > u.MaxLength {
+		slog.Warn("min length is greater than max length, falling back on default behavior")
+		return u.defaultUserPassword(username)
+	}
+
+	passLength := rand.IntN(u.MaxLength-u.MinLength) + u.MinLength
+	res, err := password.Generate(passLength, 1, 1, false, false)
+	if err != nil {
+		slog.Warn("unable to generate a proper random password, falling back on default password pattern",
```

Comment:
Under what circumstances is generating a password going to fail? This feels like possibly extra complication / unnecessary? If there's a reason like a setting where someone can set min length > max length or something that would cause it to fail, we should detect and fatal on that earlier I think.

Reply:
Yes, those values are coming from a configuration file.  So they could set say a negative min or a max of 65K or what not that probably will cause the password generator to return an error.

The two patterns are to take a sha256 of the login.json (ie file name) or create something random that won't be deterministic

Reply:
The alternative would be to just critically fail the entire process.  Rather than say do a best effort

Reply:
My gut feeling here is that having a single approach is better - make this approach deterministic. It's also simply fewer code paths, things to test, things that can break, etc. If we can't make a password because someone set it up wrong, that's on them.

Reply:
Maybe.  There's benefits to both.  Random means you will only ever see the list once when you create them.  You need to save that info otherwise it's lost.  

The previous pattern was a sha256 of the login.json which is deterministic but that also means it's a security vulnerability if someone doesn't change the passwords.  So having a feature to allow for random passwords is better in that regard.  I think both approaches are valid, it just depends on your use case/need.